clients/src/main/java/org/apache/kafka/clients/consumer/internals/DefaultBackgroundThread.java

```diff
@@ -145,19 +166,46 @@ void runOnce() {
             this.inflightEvent = Optional.empty();
         }
 
+        // TODO: Add a condition here, like shouldFindCoordinator in the future.  Since we don't always need to find
+        //  the coordinator.
+        if (coordinatorUnknown()) {
```

Comment:
nit: I'd suggest moving this check into `CoordinatorManager`. We could change the name from `tryFindCoordinator` to `maybeFindCoordinator`.

Reply:
hmm, but I think we do want to check the connection against the networkClient here.  Given that we want to decouple the coordinatorManager from the networkClient.